### PR TITLE
Core provider update

### LINKED ISSUE #3919: Update Core Terraform Provider Versions
Updating the provider versions for all core resources.

### PR DESCRIPTION
**Pull request by tim-allen-ck**
# Resolves #3919

### CODE REVIEW

**Review by marrobi — APPROVED** [review]
LGTM

File: core/terraform/.terraform.lock.hcl
Comment (marrobi) [inline core/terraform/.terraform.lock.hcl:1]: Why are you removing the lock file? Think is good to ensure the provider binaries actually match?

Comment (tim-allen-ck) [reply core/terraform/.terraform.lock.hcl:1]: added back in

**Review by Danny-Cooke-CK — APPROVED** [review]

**Pull request merged**